### ISSUE TITLE
Feature Request: Top Level Generic Widget

### ISSUE THREAD

**Issue by felipetesc** [slint-ui/slint#5178]
Feature Request: Create an abstract top level widget

Description: Currently we don't have an out of the box way to create top level widgets.  Because of this, menubars, alert boxes and other top level components must to be created for each project, and this feature seems quite relevant to everyone with the desire to use Slint. The only widget with a close behavior is AboutSlint, which can't have @children.

While searching for templates, examples and ideas about top level ui elements I found this links: 
 - https://github.com/slint-ui/slint/pull/2094
 - https://github.com/slint-ui/slint/discussions/647
 - https://codebrowser.dev/slint/slint/internal/compiler/tests/syntax/basic/popup.slint.html

We could have a component called TopLevel with resamblance and functionality from Rectangle but with the addition of another property such as "z-index", or layer-order.

Sample:
 ```slint:
export component Menu inherits TopLevel{
      x: 32px;
      y: 5px;
      layer: 1;
      
      // children widgets
}
```
Platform 1: 
OS: Windows 
Version: 11
Arch: x64 
Platform 2: 
OS: Arch Linux 
Version: KDE Plasma 6
Arch: x64 

Thanks!

**Comment by ogoffart** [comment]
You could use `Rectangle` for that, right?
Also you can add `@children` in your own component.

eg:
```
export component TopLevel { @children } 
```

**Comment by ogoffart** [comment]
I see, So you're looking at something like PopupWindow then.

**Comment by felipetesc** [comment]
> I see, So you're looking at something like PopupWindow then.

Yes, PopupWindow, as part of the default list of widgets, can do it all it's needed to the creation of menubars, dock panels, draggable widgets, dragging regions, droppable areas, popups, message boxes, touch and drag areas.  Since, this library can be used with MCU(s), softbuffer (that's really great), that will bring a great flexibility. Why not to use z since it's already done ? Rust is faster than Slint, and pass this functionality to core of the Slint render runtime will allow faster drawing too. If we team up with UI designers this brings an over simplified way to achieve better results faster. I always think about how to create menus with css, for example. Unordered list for menus?That's wrong !

Again, thanks.

**Comment by ogoffart** [comment]
Thanks.
I'll close this issue since this doesn't seem like something that is actionable. (PopupWindow has many bugs bug we have already issues for them)
